e2e/validators.go

```diff
@@ -974,3 +974,86 @@ func ValidateEnableNvidiaResource(ctx context.Context, s *Scenario) {
 	s.T.Logf("waiting for Nvidia GPU resource to be available")
 	waitUntilResourceAvailable(ctx, s, "nvidia.com/gpu")
 }
+
+func ValidateNvidiaDevicePluginServiceRunning(ctx context.Context, s *Scenario) {
+	s.T.Helper()
+	s.T.Logf("validating that NVIDIA device plugin systemd service is running")
+	
+	command := []string{
+		"set -ex",
+		"systemctl is-active nvidia-device-plugin.service",
+		"systemctl is-enabled nvidia-device-plugin.service",
+	}
+	execScriptOnVMForScenarioValidateExitCode(ctx, s, strings.Join(command, "\n"), 0, "NVIDIA device plugin systemd service should be active and enabled")
+}
+
+func ValidateNodeAdvertisesGPUResources(ctx context.Context, s *Scenario) {
+	s.T.Helper()
+	s.T.Logf("validating that node advertises GPU resources")
+	
+	// First, wait for the nvidia.com/gpu resource to be available
+	waitUntilResourceAvailable(ctx, s, "nvidia.com/gpu")
+	
+	// Get the node using the Kubernetes client from the test framework
+	nodeName := s.Runtime.KubeNodeName
+	node, err := s.Runtime.Cluster.Kube.Typed.CoreV1().Nodes().Get(ctx, nodeName, metav1.GetOptions{})
+	require.NoError(s.T, err, "failed to get node %q", nodeName)
+	
+	// Check if the node advertises GPU capacity
+	gpuCapacity, exists := node.Status.Capacity["nvidia.com/gpu"]
+	require.True(s.T, exists, "node should advertise nvidia.com/gpu capacity")
+	
+	gpuCount := gpuCapacity.Value()
+	require.Greater(s.T, gpuCount, int64(0), "node should advertise at least 1 GPU, but got %d", gpuCount)
+	
+	s.T.Logf("node %s advertises %d nvidia.com/gpu resources", nodeName, gpuCount)
+}
+
+func ValidateNvidiaDevicePluginBinaryInstalled(ctx context.Context, s *Scenario) {
+	s.T.Helper()
+	s.T.Logf("validating that NVIDIA device plugin binary is installed")
+	
+	command := []string{
+		"set -ex",
+		"dpkg -l | grep nvidia-device-plugin || rpm -qa | grep nvidia-device-plugin",
+	}
```

Comment:
Can we do this instead for more resilient code, so we know this is failing reliably and not because of the side effect that `dpkg not found` or `rpm not found` on the other OSes:


```suggestion
    command := []string{"set -ex"}

	switch s.Config.VHD.OS {
	case config.OSAzureLinux:
		command = append(command, "rpm -qa | grep nvidia-device-plugin")
	case config.OSUbuntu:
		command = append(command, "dpkg -l | grep nvidia-device-plugin")
	default:
		s.T.Fatalf("unsupported OS for NVIDIA device plugin validation: %s", s.Config.VHD.OS)
	}
```